
\documentclass[hyperref={pdfpagelabels=false}, compress]{beamer}
%\let\Tiny=\tiny

\usepackage[T1]{fontenc} 
\usepackage{lmodern}
\usepackage{graphicx}
\usepackage{hyperref}
\usepackage{url}
%\usepackage[style=footnote-dw]{biblatex}
%\usepackage[style=apa]{biblatex}
\usepackage{BeamerColor}

%\usepackage{natbib}
%\bibliographystyle{apalike}

%\mode<presentation>
%{  
	\usetheme{Frankfurt}
  	\usecolortheme[named=firebrick4]{structure}
	%\usecolortheme{beaver}
	%\usecolortheme{lily}
  	\useinnertheme{circles}
  	\usefonttheme[onlymath]{serif}
  	\setbeamercovered{transparent}
  	\setbeamertemplate{blocks}[rounded][shadow=true]
	\setbeamertemplate{footline}[page number]{}
	\setbeamertemplate{navigation symbols}{}
	%\setbeamerfont{footnote}{size=\tiny}
%}

%\bibliography{References} 

\newenvironment{changemargin}[2]{%
  \begin{list}{}{%
    \setlength{\topsep}{0pt}%
    \setlength{\leftmargin}{#1}%
    \setlength{\rightmargin}{#2}%
    \setlength{\listparindent}{\parindent}%
    \setlength{\itemindent}{\parindent}%
    \setlength{\parsep}{\parskip}%
  }%
  \item[]}{\end{list}}

\renewcommand{\familydefault}{\sfdefault}

%%%%%%%%%%

\title{Dictybase overhaul}
\author{\textit{dictyBase Developers}}

%\date[October 17, 2012]
\date

\begin{document}

\frame{\titlepage}

%\section[Overview]{}
%\frame{\tableofcontents}

%%%%%%%%%%

\section{}
\begin{frame}
	\frametitle{Proposed migration tasks}
	
    	\begin{itemize}
		\item \textbf{ Data migration from Oracle to PostgreSQL [ETA: 2 - 3 months)}
    		\item Develop Modware
    		\item Rewrite existing web application
    		\item Migrate CGI scripts to web applications 
    		\item Rewrite standalone scripts and pipelines
    		\item Behind the scene projects (Unit testing/Deployment etc ..)
    		\item Third party tools

	\end{itemize}
		
\end{frame}

\begin{frame}
    \frametitle{Data Migration - Approaches}
       \begin{itemize}
           \item \textbf{\huge Direct} - {\Large Inter Database SQL based}
           \begin{enumerate}
              \item Manipulate existing data models in Oracle
              \item Bulk SQL based transfer to PostgreSQL
           \end{enumerate}
           \item \textbf{\huge Indirect} - {\Large Flat file based}
            \begin{enumerate}
               \item Dumps data from Oracle to std/nonstd flat file format
               \item Load in PostgreSQL via loader scripts
            \end{enumerate}
       \end{itemize}
\end{frame}

\begin{frame}
    \frametitle{Data Migration - Direct Approach}
       \begin{itemize}
          \item  \textbf{\Large Manipulate data models}
           \begin{itemize}
               \item Core gene model [SB]
               \item Core gene model annotations [YP]
               \item Core ontologies \textit{(Excluding GO)} [SB]
               \item Phenotype, genotype and strain [YP]
               \item Protein domains
               \item Homolog/Ortholog
               \item Regulatory element/Promoter
           \end{itemize}
          \end{itemize}
\end{frame}


\begin{frame}
    \frametitle{Direct approach - Data transfer}  
          \begin{itemize}
               \item Via SQL dump [SB and YP]
               \item Via Foreign Data Wrappers(FDW) without SQL dump [SB and YP]
           \end{itemize} 
         \textbf{\large Unfamiliar ground, needs more leg work to flesh out a clear path}
\end{frame}

\begin{frame}
    \frametitle{Direct approach - Problems ??}  
         \textbf{\Large Jugging with data model}
          \begin{itemize}
               \item Almost impossible to update SO(Sequence ontology)
               \item Updating ontologies (PATO,phonotype,anatomy etc ...) ??
               \item Migrating half  baked strains data model.  Could be Non-trivial and quite complicated.
           \end{itemize} 
\end{frame}



\begin{frame}
    \frametitle{Data Migration - Indirect Approach}
       \begin{itemize}
          \item  \textbf{\Large Export to flat file}
           \begin{itemize}
               \item Core gene models as GFF3 [SB]
               \item Core gene model annotations [YP]
               \item Literature annotations [SB]
               \item Phenotype, genotype and strain data and annotations [YP]
           \end{itemize}
          \end{itemize}
\end{frame}


\begin{frame}
  \frametitle{Indirect Approach - Data loading}
  
  \begin{itemize}
       \item Ontologies from obo [SB]
       \item GFF3 loading [SB]
       \item Gene annotations,Literature annotations [SB/YP]
       \item GAF loading [YP]
       \item Phenotype,genotype and strains [YP]
       \item Protein domains from Interpro flat file
       \item Homologs from flat file
  \end{itemize}
  \textbf{\Large Familiar ground, less leg work}
  \end{frame}
  
  \begin{frame}
    \frametitle{Indirect approach - Problems ??}  
         \textbf{\Large Nothing potentially serious or obvious at this point}
 \end{frame}

\begin{frame}
    \frametitle{Data Migration - Data Models to consider}
           \begin{itemize}
               \item UTRs
               \item User model (Curators contributions to annotations)
              \end{itemize}        
\end{frame}

\begin{frame}
   \frametitle{Data Migration - Maintenance}
   \textbf{\Huge PostgreSQL DBA?}
\end{frame}

%%%%%%%%%%

\end{document}
